pkgs/applications/science/math/sage/sage-src.nix

```diff
@@ -8,87 +8,45 @@
-      # unrelated changes. If there is no such tag (due to nonlinear history, for
-      # example), there are two other options, listed below.
-      #
-      # 2) From GitHub's sagemath/sage repo. This lets us use a GH feature that allows
-      # us to choose between a .patch file, with one patch per commit, or a .diff file,
-      # which squashes all commits into a single diff. This is used if squashed ==
-      # true. This repo has all release tags. However, it has no developer branches, so
-      # this option can't be used if a change wasn't yet shipped in a (possibly beta)
-      # release.
-      #
-      # 3) From GitHub's sagemath/sagetrac-mirror repo. Mirrors all developer branches,
-      # but has no release tags. The only use case not covered by 1 or 2 is when we need
-      # to apply a patch from an open ticket that contains merge commits.
-      #
-      # Item 3 could cover all use cases if the sagemath/sagetrack-mirror repo had
-      # release tags, but it requires a sha instead of a release number in "base", which
-      # is inconvenient.
-      urls = if squashed
-             then [
-               "https://github.com/sagemath/sage/compare/${base}...${rev}.diff"
-               "https://github.com/sagemath/sagetrac-mirror/compare/${base}...${rev}.diff"
-             ]
-             else [ "https://git.sagemath.org/sage.git/patch?id2=${base}&id=${rev}" ];
-    } // builtins.removeAttrs args [ "rev" "base" "sha256" "squashed" "excludes" ])
-  );
-in
 stdenv.mkDerivation rec {
-  version = "9.8";
+  version = "10.0";
   pname = "sage-src";
 
   src = fetchFromGitHub {
     owner = "sagemath";
     repo = "sage";
     rev = version;
-    sha256 = "sha256-dDbrzJXsOBARYfJz0r7n3LbaoXHnx7Acz6HBa95NV9o=";
+    sha256 = "sha256-zN/Lo/GBCjYGemuaYpgG3laufN8te3wPjXMQ+Me9zgY=";
   };
 
   # Patches needed because of particularities of nix or the way this is packaged.
   # The goal is to upstream all of them and get rid of this list.
   nixPatches = [
-    # Fixes a potential race condition which can lead to transient doctest failures.
-    ./patches/fix-ecl-race.patch
-
-    # Not necessary since library location is set explicitly
-    # https://trac.sagemath.org/ticket/27660#ticket
-    ./patches/do-not-test-find-library.patch
-
     # Parallelize docubuild using subprocesses, fixing an isolation issue. See
     # https://groups.google.com/forum/#!topic/sage-packaging/YGOm8tkADrE
     ./patches/sphinx-docbuild-subprocesses.patch
+
+    # After updating smypow to (https://github.com/sagemath/sage/issues/3360)
+    # we can now set the cache dir to be within the .sage directory. This is
+    # not strictly necessary, but keeps us from littering in the user's HOME.
+    ./patches/sympow-cache.patch
+
+    # We don't run the documentation bootstrap script, so we don't have a list
+    # of Sage-the-distribution SPKGs. Remove doc references to them.
+    ./patches/dont-reference-spkg.patch
```

Comment:
I am considering just running `src/doc/bootstrap`, it would remove a few hacks (and it doesn't require running `src/bootstrap`, which is good).

Reply:
Are there any drawbacks to using that `bootstrap` script?

Reply:
The root bootstrap script requires network, so it is a no go. The doc bootstrap script, which generates a page of documentation for each external package required by Sage, expects to be called from the root one and reuses some of its infrastructure to list packages, but with a couple of hacks it runs fine.

The main drawback is conceptual: it uses spkg information from Sage-the-distribution (the `build` directory), but it's not too bad since it's just package names and versions which are contained in the main Sage repository. It's just a bit unfortunate that part of the upstream documentation (the section on how to use the GitHub CLI) now has links to the spkg section of the manual, requiring us to either not ignore the `build` directory or patch the references away.

I've pushed a commit that runs the doc bootstrap script (but not the root one). The previous version of the update just manually patched the (single) spkg reference away from the developer documentation. Either solution is fine by me.